### ISSUE TITLE
Incorrect slot value on head event?

### ISSUE THREAD

**Issue by mcdee** [Consensys-Incorporated/teku#4747]
I have been investigating an issue where some blocks are received extremely late but a node, in to the following slot.  An example of this on mainnet is slot 2651855, which was received around 13 seconds after the start of its slot.  The following block, 2651856, was received around 500ms after the start of its slot, so notably _before_ the block for the prior slot.

The teku head events from the `/events` endpoint generated in this situation were:

```
event:head
data:{"slot":"2651856","block":"0xbd76e4a6163720e526b8c44a249a2a4b663cae9074d8cc3f82689a38f1637d26","state":"0x46f581ff0c92647dee09d513eb93aa4a75c8eb2149baab01770403b6229d910a","epoch_transition":false,"previous_duty_dependent_root":"0x3f8becd864124efa4acede4fe461e028a1be0ff3d73d2bc0c27ecbf233fe1777","current_duty_dependent_root":"0x8f35304bed23213ede76021b0847177f0f29387fa8baa0d8a8cab67c4120c4ba"}

event:head
data:{"slot":"2651856","block":"0xffe1c9fef22e4f15eeac3e6c8756f261d18ab86733628f17148a4b504bdd1542","state":"0xa582b8720a7cd5f13c4c0e5e28d1e0ea5db7902225cba08a21a4d503fb5b6518","epoch_transition":false,"previous_duty_dependent_root":"0x3f8becd864124efa4acede4fe461e028a1be0ff3d73d2bc0c27ecbf233fe1777","current_duty_dependent_root":"0x8f35304bed23213ede76021b0847177f0f29387fa8baa0d8a8cab67c4120c4ba"}
```

All of this data ties up with information from other nodes, except the `slot` number.  prysm and lighthouse show the slot number for the second event as `2651855`.

Now, it could be argued that `slot` should be the slot of the chain, which appears to be what teku is showing, or that it should be the slot of the head block, which appears to be what prysm and lighthouse are showing.  Given that teku is currently in the minority I'm raising an issue here for discussion (but personally think that teku's use of slot is more accurate/useful).

**Comment by ajsutton** [comment]
Yeah I'd say Teku is being weird here. We're reporting what we call the "fork choice slot" which is basically a vestige of the fact that, prior to dependentRoots being added to the API, Teku would consider a slot "officially empty" 4s into the slot and publish a reorg event if it was fill after that point. To do this, we track the latest slot that fork choice has selected the head for which is the "fork choice slot". 

I think reporting the block slot is more intuitive and likely safer in terms of ensuring that validator clients correctly identify when duties should be invalidated. So I'd say we need to fix Teku.  Someone receiving the event could always independently calculate the slot based on current time which is pretty much what Teku is reporting.

**Comment by rolfyone** [comment]
This issue is resolved by #4749
